components/org.wso2.carbon.identity.query.saml/src/test/java/org/wso2/carbon/identity/query/saml/validation/SAMLSubjectQueryValidatorTest.java

```diff
@@ -0,0 +1,312 @@
+        DummyIssuer issuer3 = new DummyIssuer();
+        DummyIssuer issuer4 = new DummyIssuer();
+        DummyIssuer issuer5 = new DummyIssuer();
+
+        DummyIssuer issuer6 = new DummyIssuer();
+        issuer1.setValue("test");
+        issuer2.setValue("test");
+        issuer3.setValue("test");
+        issuer4.setValue("test");
+        issuer5.setValue("test");
+        issuer6.setValue("test");
+
+        issuer1.setFormat(SAMLQueryRequestConstants.GenericConstants.ISSUER_FORMAT);
+        issuer2.setFormat(SAMLQueryRequestConstants.GenericConstants.ISSUER_FORMAT);
+        issuer3.setFormat(SAMLQueryRequestConstants.GenericConstants.ISSUER_FORMAT);
+        issuer4.setFormat(SAMLQueryRequestConstants.GenericConstants.ISSUER_FORMAT);
+        issuer5.setFormat(SAMLQueryRequestConstants.GenericConstants.ISSUER_FORMAT);
+        issuer6.setFormat(SAMLQueryRequestConstants.GenericConstants.ISSUER_FORMAT);
+
+        DummySubjectQuery dumSQ1 = new DummySubjectQuery();
+        DummySubjectQuery dumSQ2 = new DummySubjectQuery();
+        DummySubjectQuery dumSQ3 = new DummySubjectQuery();
+        DummySubjectQuery dumSQ4 = new DummySubjectQuery();
+        DummySubjectQuery dumSQ5 = new DummySubjectQuery();
+        DummySubjectQuery dumSQ6 = new DummySubjectQuery();
+
+        dumSQ2.setSubject(dumSub1);
+        dumSQ3.setSubject(dumSub2);
+        dumSQ4.setSubject(dumSub3);
+        dumSQ5.setSubject(dumSub4);
+
+        dumSQ1.setIssuer(issuer1);
+        dumSQ2.setIssuer(issuer2);
+        dumSQ3.setIssuer(issuer3);
+        dumSQ4.setIssuer(issuer4);
+        dumSQ5.setIssuer(issuer5);
+        dumSQ6.setIssuer(issuer6);
+
+        dumSQ1.setVersion(SAMLVersion.VERSION_20);
+        dumSQ2.setVersion(SAMLVersion.VERSION_20);
+        dumSQ3.setVersion(SAMLVersion.VERSION_20);
+        dumSQ4.setVersion(SAMLVersion.VERSION_20);
+        dumSQ5.setVersion(SAMLVersion.VERSION_20);
+        dumSQ6.setVersion(SAMLVersion.VERSION_10);
+
+        SAMLSSOServiceProviderDO ssoIdpConfigs1 = new SAMLSSOServiceProviderDO();
+        SAMLSSOServiceProviderDO ssoIdpConfigs2 = new SAMLSSOServiceProviderDO();
+        ssoIdpConfigs1.setCertAlias("test");
+        ssoIdpConfigs1.setAssertionQueryRequestProfileEnabled(true);
+
+        ssoIdpConfigs2.setNameIDFormat("test");
+        ssoIdpConfigs2.setCertAlias("test");
+        ssoIdpConfigs2.setAssertionQueryRequestProfileEnabled(true);
+        return new Object[][]{
+                {dumSQ1, false, ssoIdpConfigs2},
+                {dumSQ6, false, ssoIdpConfigs2},
+                {dumSQ2, false, ssoIdpConfigs2},
+                {dumSQ3, false, ssoIdpConfigs2},
+                {dumSQ4, false, ssoIdpConfigs1},
+                {dumSQ5, false, ssoIdpConfigs2},
+                {dumSQ4, false, ssoIdpConfigs2},
+                {dumSQ4, true, ssoIdpConfigs2}
+        };
+    }
+
+    @Test(dataProvider = "provideSubectQuery")
+    public void testValidate(Object SubQ, boolean value, Object ssoP)
```

Comment:
use meaning parameter names: 'value' and 'ssoP'

Reply:
please fix.

Reply:
fixed